packages/core/core/src/providers/sessionManager.ts

```diff
@@ -0,0 +1,34 @@
+import { defineProvider } from './provider';
+import { createSessionManager } from '../services/session-manager';
+
+interface AdminAuthConfig {
+  secret?: string;
+  options?: Record<string, unknown>;
+}
+
+export default defineProvider({
+  init(strapi) {
+    // Get JWT secret from admin auth settings (same as admin token service)
+    const adminAuth = strapi.config.get<AdminAuthConfig>('admin.auth', {});
+    const jwtSecret = adminAuth.secret;
+
+    if (!jwtSecret) {
+      throw new Error(
+        'Missing admin.auth.secret configuration. The SessionManager requires a JWT secret'
+      );
+    }
+
+    const config = {
+      jwtSecret,
+      refreshTokenLifespan: 30 * 24 * 60 * 60, // 30 days in seconds
```

Comment:
We should make these configurable, similar to what we do with the rate-limit middleware

Reply:
➕

Reply:
but if that's coming in a follow-up PR, that's fine, we don't have to address it here